lib/features/buildings_view/model/building_model.dart

```diff
@@ -14,11 +19,46 @@ class BuildingModel extends Building implements GoogleNavigable {
           cover: building.cover,
           naturalName: building.naturalName,
           disableBuildingPrefix: building.disableBuildingPrefix,
-        );
+        ) {
+    _buildingCodesLowerCase.addBuildingCode(name);
+  }
 
   @override
   LatLng get location => LatLng(latitude, longitude);
 
   String? get addressFormatted =>
       addres?.replaceFirst(",", "\n").replaceAll("\n ", "\n");
+
+  static bool isBuildingCode(String buildingCode) {
+    return _buildingCodesLowerCase.contains(buildingCode) ||
+        buildingCode.containsNumber();
+  }
+}
+
+class BuildingCodesLowerCase {
```

Comment:
Sorry to hurt your feelings, but I'm not a biggest fan of this approach 😭😭 I would highly recomend doing this in some functional programming manner instead. 

The logic itself looks fine, I'm just not a biggest fan of this static class. This could be much cleaner and shorter using more declarative and less imperative paradigm. Maybe we could use the benefits of riverpod here too?

I can give you some example code in a second